api/models/User.js

```diff
@@ -1317,7 +1317,8 @@ UserSchema.methods = {
 
   backupCodeIndex(code) {
     let index = -1;
-    for (let i = 0; i < this.totpConf.backupCodes.length; i += 1) {
+    let numCodes = this.totpConf && this.totpConf.backupCodes && this.totpConf.backupCodes.length;
+    for (let i = 0; i < numCodes; i++) {
```

Comment:
@orakili upon looking at this after sleeping, I thought it good to call out the three lines of code you're auditing 😁  this is one spot.